articles/active-directory/develop/tutorial-blazor-server.md

```diff
@@ -34,173 +34,70 @@ In this tutorial:
 
-In your browser, navigate to `https://localhost:<port number> `, and log in using an Azure AD user account to see the app running.
+> For more information on user secrets storage please visit: [safe storage of app secrets during development](/aspnet/core/security/app-secrets)
 
-## Retrieving data from Microsoft Graph
+## Use the Microsoft Identity App Sync Tool
 
-[Microsoft Graph](/graph/overview) offers a range of APIs that provide access to your users' Microsoft 365 data. By using the Microsoft identity platform as the identity provider for your app, you have easier access to this information since Microsoft Graph directly supports the tokens issued by the Microsoft identity platform. In this section, you add code to display the signed in user's emails on the application's "fetch data" page.
+Run the following command to register your app in your tenant and update the .Net configuration of your application. Provide the username/upn belonging to your Azure Account (for instance username@domain.com) and the tenant id or domain name of the Azure Active Directory associated with your Azure Account. If you use an account that is signed-in in either Visual Studio, Azure CLI, or Azure PowerShell, you'll benefit from Single Sign On (SSO).
 
-Before you start, log out of your app since you'll be making changes to the required permissions, and your current token won't work. If you haven't already, run your app again and select **Log out** before updating the code below.
+```dotnetcli
+msidentity-app-sync --username <username/upn> --tenant-id <tenantID>
+```
 
-Now you'll update your app's registration and code to pull a user's email and display the messages within the app. To achieve this, first extend the app registration permissions in Azure AD to enable access to the email data. Then, add code to the Blazor app to retrieve and display this data in one of the pages.
+> [!Note]
+>
+> - you don't need to provide the username if you are signed-in with only one account in the developer tools.
+> - you don't need to provide the tenant-id if the tenant in which you want to create the application is your home tenant.
 
-1. In the Azure portal, select your app in **App registrations**.
-1. Under **Manage**, select **API permissions**.
-1. Select **Add a permission** > **Microsoft Graph**.
-1. Select **Delegated Permissions**, then search for and select the **Mail.Read** permission.
-1. Select **Add permissions**.
+## Optional - Create a development SSL certificate
 
-In the _appsettings.json_ file, update your code so it fetches the appropriate token with the right permissions. Add `mail.read` after the `user.read` scope under `DownstreamAPI`. This is specifying which scopes (or permissions) the app will request access to.
+In order to avoid SSL errors / warning when browsing the running application you can use the following on macOS and Windows to generate a self-signed SSL certificate for use by .Net core.
```

Comment:
```suggestion
In order to avoid SSL errors/warnings when browsing the running application, you can use the following on macOS and Windows to generate a self-signed SSL certificate for use by .NET Core.
```